File: apk/src/test/java/dex/DexClassTypeToStringTest.java
package dex;

import org.junit.Assert;
import org.junit.Test;
import uk.co.alt236.apk.repo.dex.DexClassTypeToString;
import uk.co.alt236.apk.repo.dex.model.DexClassType;

import static org.junit.Assert.assertEquals;

public class DexClassTypeToStringTest {

    @Test
    public void convertsVoidsProperly() {
        final String typeString = "V";
        final String typeName = "void";

        final DexClassType classType = new DexClassType(typeString);
        Assert.assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertsBoolProperly() {
        final String typeString = "Z";
        final String typeName = "bool";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertsByteProperly() {
        final String typeString = "B";
        final String typeName = "byte";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }


    @Test
    public void convertsObjectProperly() {
        final String typeString = "Ljava/lang/Throwable;";
        final String typeName = "Ljava/lang/Throwable;";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertsIntProperly() {
        final String typeString = "I";
        final String typeName = "int";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertsLongProperly() {
        final String typeString = "J";
        final String typeName = "long";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertsShortProperly() {
        final String typeString = "S";
        final String typeName = "short";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertsFloatProperly() {
        final String typeString = "F";
        final String typeName = "float";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertsCharProperly() {
        final String typeString = "C";
        final String typeName = "char";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertsDoubleProperly() {
        final String typeString = "D";
        final String typeName = "double";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertsInitProperly() {
        final String typeString = "<init>";
        final String typeName = "<init>";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertsClinitProperly() {
        final String typeString = "<clinit>";
        final String typeName = "<clinit>";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertOneDimensionBoolArray() {
        final String typeString = "[Z";
        final String typeName = "bool[]";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertTwoDimensionBoolArray() {
        final String typeString = "[[Z";
        final String typeName = "bool[][]";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }


    @Test
    public void convertOneDimensionObjectArray() {
        final String typeString = "[Ljava/lang/Throwable;";
        final String typeName = "Ljava/lang/Throwable;[]";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }

    @Test
    public void convertTwoDimensionObjectArray() {
        final String typeString = "[[Ljava/lang/Throwable;";
        final String typeName = "Ljava/lang/Throwable;[][]";

        final DexClassType classType = new DexClassType(typeString);
        assertEquals(typeName, DexClassTypeToString.toString(classType));
    }


}

File: apk/src/test/java/dex/DexClassTypeToStringTest.kt
package dex

import org.junit.Assert
import org.junit.Assert.assertEquals
import org.junit.Test
import uk.co.alt236.apk.repo.dex.DexClassTypeToString
import uk.co.alt236.apk.repo.dex.model.DexClassType

class DexClassTypeToStringTest {

    @Test
    fun convertsVoidsProperly() {
        val typeString = "V"
        val typeName = "void"

        val classType = DexClassType(typeString)
        Assert.assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertsBoolProperly() {
        val typeString = "Z"
        val typeName = "bool"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertsByteProperly() {
        val typeString = "B"
        val typeName = "byte"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }


    @Test
    fun convertsObjectProperly() {
        val typeString = "Ljava/lang/Throwable;"
        val typeName = "Ljava/lang/Throwable;"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertsIntProperly() {
        val typeString = "I"
        val typeName = "int"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertsLongProperly() {
        val typeString = "J"
        val typeName = "long"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertsShortProperly() {
        val typeString = "S"
        val typeName = "short"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertsFloatProperly() {
        val typeString = "F"
        val typeName = "float"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertsCharProperly() {
        val typeString = "C"
        val typeName = "char"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertsDoubleProperly() {
        val typeString = "D"
        val typeName = "double"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertsInitProperly() {
        val typeString = "<init>"
        val typeName = "<init>"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertsClinitProperly() {
        val typeString = "<clinit>"
        val typeName = "<clinit>"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertOneDimensionBoolArray() {
        val typeString = "[Z"
        val typeName = "bool[]"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertTwoDimensionBoolArray() {
        val typeString = "[[Z"
        val typeName = "bool[][]"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }


    @Test
    fun convertOneDimensionObjectArray() {
        val typeString = "[Ljava/lang/Throwable;"
        val typeName = "Ljava/lang/Throwable;[]"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }

    @Test
    fun convertTwoDimensionObjectArray() {
        val typeString = "[[Ljava/lang/Throwable;"
        val typeName = "Ljava/lang/Throwable;[][]"

        val classType = DexClassType(typeString)
        assertEquals(typeName, DexClassTypeToString.toString(classType))
    }


}